### ISSUE TITLE
`if_clean &block` leads to scripts exiting zero even if they didn't perform their action.

### ISSUE THREAD

**Issue by rib3** [sportngin/octopolo#64]
With a dirty index, `op stage-up && deploy` will run `deploy` even though it won't stage anything up because `if_clean &block` doesn't trigger an error.

The easiest fix would be to add an `exit 1` after the `alert_dirty_index` call [here](https://github.com/sportngin/octopolo/blob/c5cb5c11278f4b94bb9d3d56e2c2db475485a058/lib/octopolo/git.rb#L115)

I personally think something like a `check_status` method that doesn't take a block makes more sense for scripts like this.  The method would throw an exception that bubbles up to the user and causes the script to exit (non zero). ...or the exception could be caught and handled.

```
def perform
  git.check_status
  check_out_branch
  merge_pull_request
  comment_about_merge
end
```
vs:
```
def perform
  git.if_clean do
    check_out_branch
    merge_pull_request
    comment_about_merge
  end
end
```

**Comment by emmasax1** [comment]
Personally, I think that you should feel free to add the `exit 1` piece after the `alert_dirty_index`. I haven't experienced this problem specifically, but if it'll help, it could very well be worth it.

**Comment by anfleene** [comment]
failing should exit non zero. This is a bug.

**Comment by astuyve** [comment]
<3 @rib3 fixed in #81 will let you know when the next release is out.
